test/unittests/frontend.js

```diff
@@ -0,0 +1,64 @@
+/*
```

Comment:
How about we adopt the convention that each source file has a corresponding test file? Since these tests are for notebook-list.js, this file should be named that, or test-notebook-list.js, or something like that.

Also, will these tests be for both the ts and the js files? I.e. for both the notebook server and the client code? Do we want to put those into separate directories?